pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -62,14 +62,30 @@ const (
 
 // MigClusterSpec defines the desired state of MigCluster
 type MigClusterSpec struct {
+	// Specifies if the cluster is host (where the controller is installed) or not. This is a required field.
 	IsHostCluster           bool                  `json:"isHostCluster"`
+
+	// Stores the url of the remote cluster. The field is only required for the source cluster object.
 	URL                     string                `json:"url,omitempty"`
+
 	ServiceAccountSecretRef *kapi.ObjectReference `json:"serviceAccountSecretRef,omitempty"`
+
+	// If the migcluster needs SSL verification for connections a user can supply a custom CA bundle.
```

Comment:
We should specify that the only time this may be required is when `spec.Insecure = false`